scripts/channel_map.py

```diff
@@ -0,0 +1,96 @@
+            'branch': 'release/3.1'
+        },
+        '3.0': {
+            'tfm': 'netcoreapp3.0',
+            'branch': 'release/3.0'
+        },
+        '2.2': {
+            'tfm': 'netcoreapp2.2',
+            'branch': 'release/2.2'
+        },
+        '2.1': {
+            'tfm': 'netcoreapp2.1',
+            'branch': 'release/2.1'
+        },
+        'LTS': {
+            'tfm': 'net461', # For Full Framework download the LTS for dotnet cli.
+            'branch': 'LTS'
+        }
+    }
+    @staticmethod
+    def get_supported_channels() -> list:
+        '''List of supported channels.'''
+        return list(ChannelMap.channel_map.keys())
+
+    @staticmethod
+    def get_supported_frameworks() -> list:
+        '''List of supported frameworks'''
+        frameworks = [ChannelMap.channel_map[channel]['tfm'] for channel in ChannelMap.channel_map]
+        return set(frameworks)
+
+    @staticmethod
+    def get_branch(channel: str) -> str:
+        if channel in ChannelMap.channel_map:
+            return ChannelMap.channel_map[channel]['branch']
+        else:
+            raise Exception('Channel %s is not supported. Supported channels %s' % (channel, ChannelMap.get_supported_channels()))
+
+    @staticmethod
+    def get_target_framework_monikers(channels: list) -> list:
+        '''
+        Translates channel names to Target Framework Monikers (TFMs).
+        '''
+        monikers = [
+            ChannelMap.get_target_framework_moniker(channel)
+            for channel in channels
+        ]
+        return list(set(monikers))
+
+    @staticmethod
+    def get_target_framework_moniker(channel: str) -> str:
+        '''
+        Translate channel name to Target Framework Moniker (TFM)
+        '''
+        if channel in ChannelMap.channel_map:
+            return ChannelMap.channel_map[channel]['tfm']
+        else:
+            raise Exception('Channel %s is not supported. Supported channels %s' % (channel, ChannelMap.get_supported_channels()))
+
+    @staticmethod
+    def get_channel_from_target_framework_moniker(target_framework_moniker: str) -> str:
+        '''Translate Target Framework Moniker (TFM) to channel name'''
+        for channel in ChannelMap.channel_map:
+            if ChannelMap.channel_map[channel]['tfm'] == target_framework_moniker:
+                return channel
+        raise Exception('Framework %s is not supported. Supported frameworks: %s' % (target_framework_moniker, ChannelMap.get_supported_frameworks()))
+
+
+def __main():
+    parser = ArgumentParser()
+    parser.add_argument('--channel', type=str, required=True, dest='channel', help='channel name or status')
+    parser.add_argument('--setup-pipeline', action='store_true', dest='setup_pipeline', help='channel required to set up pipeline variables')
+    args = parser.parse_args()
+
+    if args.setup_pipeline:
+        print('##vso[task.setvariable variable=_Framework]%s' % ChannelMap.get_target_framework_moniker(args.channel))
```

Comment:
it would be nice to add a comment here explaining what it is doing

Reply:
Thanks for your suggestion! Will add it in the following PR.